Audit and refactor framework settings

barrymcgee opened the pull request:
## Done

- Remove global settings that are too wide ranging in scope
- Split out remaining variables into distinct categories based on function
- Rename all variables to prefix them with their category
- Update `docs` with changes
- Update Release Notes to document changes

## QA

* Sanity check code changes
* Pull down code and run `gulp build`
* Ensure build passes

Fixes: #386

anthonydillon commented on V1_RELEASE_NOTES.md:
```diff
@@ -3,3 +3,46 @@
+
+* `$navigation-threshold` has been replaced with `$breakpoint-small`
+
+* `$asset-server` is now `$assets-server`
```
I would like to rename this one too `$assets-path` as it is used to link to local assets when the site news to work offline.

anthonydillon commented on docs/settings/_assets-settings.md:
```diff
@@ -0,0 +1,8 @@
+
+Setting  | Default Value
+ ------------- | -------------
+$assets-server:   | https://assets.ubuntu.com/v1/
```
See my note in the change log

anthonydillon commented on docs/settings/_font-settings.md:
```diff
@@ -0,0 +1,11 @@
+
+Setting  | Default Value
+ ------------- | -------------
+$font-assets-path   | 'https://assets.ubuntu.com/sites/ubuntu/latest/u/'
```
This is the old asset path. It should contain `v1`

anthonydillon commented on docs/settings/_font-settings.md:
```diff
@@ -0,0 +1,11 @@
+ ------------- | -------------
+$font-assets-path   | 'https://assets.ubuntu.com/sites/ubuntu/latest/u/'
+$font-base-family   | 'Ubuntu, Arial, "libra sans", sans-serif'
+$font-base-size   | 1rem
```
This will set the body font size to `1rem` we should set the body font size as a px value and use rem from there

anthonydillon commented:
Thanks Barry, few comments.

barrymcgee commented:
@anthonydillon I was wondering why we needed both, spoke to Graham and he explained the legacy issue. I'm now removed '$font-assets-path' entirely and updated all references to use '$assets-path'.

Ready for re-review.

anthonydillon commented:
Perfect, thank you :+1:
